### PR TITLE
Update for 0.9.1 release

### PR DESCRIPTION
**Pull request by harripd**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by harripd** [comment]
@conda-forge/core Can you help with the linter error? I am unable to find documentation defining python selectors and noarch specifications. 

> Hi! This is the friendly automated conda-forge-linting service.
> 
> I wanted to let you know that I linted all conda-recipes in your PR (`recipe`) and found some lint.
> 
> Here's what I've got...
> 
> For **recipe**:
> 
>     * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,44 +1,51 @@
...
-  script: python -m pip install -vv --no-deps --ignore-installed .
+  script:
+    - export SETUPTOOLS_SCM_PRETEND_VERSION={{ version }} 
+    - {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
+  number: 0
 
 requirements:
   host:
-    - pip
     - python
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.6
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,44 +1,51 @@
...
+    - setuptools-scm >=6.0
+    - pip
   run:
-    - python >=3.5
-    - setuptools
-    - numpy >=1.9
+    - python >=3.6
+    - numpy >=1.19
     - pytables
+    - importlib_resources  # [py<=310]
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
    - importlib_resources
```

just remove this "if statement". Conda doesn't support them in the same way that pip does.

You can have it if you want to have a huge build matrix and build out separate packages for python 3.8, 3.9. 3.10, 3.11, 3.12, and pypy, but it really isn't critical that it "not be installed" with python 3.8-3.10.

So simplify your recipe, and force users to install importlib_resorces with your package.

Comment (hmaarrfk) [reply recipe/meta.yaml]: Otherwise, with conda-forge, you would have to remove the line that says `noarch: python`.

Comment (harripd) [reply recipe/meta.yaml]: After double checking that I can still install the` importlib_resources` back-port of `importlib.resrouces` on the newer version of Python, I think that yes, forcing install is the best option.

Thank you for your help.

**Pull request merged**